services/catalog/src/simcore_service_catalog/services/access_rights.py

```diff
@@ -0,0 +1,167 @@
+""" Services Access Rights policies
+
+"""
+import logging
+from datetime import datetime
+from typing import List, Optional, Tuple
+from urllib.parse import quote_plus
+
+from aiopg.sa.engine import Engine
+from fastapi import FastAPI
+from models_library.services import ServiceAccessRightsAtDB, ServiceDockerData
+from packaging.version import Version
+from pydantic.types import PositiveInt
+
+from ..api.dependencies.director import get_director_api
+from ..db.repositories.groups import GroupsRepository
+from ..db.repositories.services import ServicesRepository
+from ..utils.versioning import as_version, is_patch_release
+
+logger = logging.getLogger(__name__)
+
+OLD_SERVICES_DATE: datetime = datetime(2020, 8, 19)
+
+
+def _is_frontend_service(service: ServiceDockerData) -> bool:
+    return "/frontend/" in service.key
+
+
+async def _is_old_service(app: FastAPI, service: ServiceDockerData) -> bool:
+    # get service build date
+    client = get_director_api(app)
+    data = await client.get(
+        f"/service_extras/{quote_plus(service.key)}/{service.version}"
+    )
+    if not data or "build_date" not in data:
+        return True
+
+    logger.debug("retrieved service extras are %s", data)
+
+    service_build_data = datetime.strptime(data["build_date"], "%Y-%m-%dT%H:%M:%SZ")
+    return service_build_data < OLD_SERVICES_DATE
+
+
+async def evaluate_default_policy(
+    app: FastAPI, service: ServiceDockerData
+) -> Tuple[Optional[PositiveInt], List[ServiceAccessRightsAtDB]]:
+    """Given a service, it returns the owner's group-id (gid) and a list of access rights following
+    default access-rights policies
+
+    - DEFAULT Access Rights policies:
+        1. All services published in osparc prior 19.08.2020 will be visible to everyone (refered as 'old service').
+        2. Services published after 19.08.2020 will be visible ONLY to his/her owner
+        3. Front-end services are have read-access to everyone
+    """
+    db_engine: Engine = app.state.engine
+
+    groups_repo = GroupsRepository(db_engine)
+
+    everyone_gid = (await groups_repo.get_everyone_group()).gid
+    owner_gid = None
+    reader_gids: List[PositiveInt] = []
```

Comment:
what means reader of a service?
What about using something different? executer_gids?

Reply:
This was already there.  I changed the name